src/raw_socket_device.c

```diff
@@ -34,8 +34,8 @@
 static const char *device_info = "raw_socket";
 
 static bool setup_device(void) {
-	struct ifreq ifr;
-	struct sockaddr_ll sa;
+	struct ifreq ifr = {0};
+	struct sockaddr_ll sa = {0};
```

Comment:
Hm, this was problematic in the past, with compilers complaining about not enough parentheses around initializers. The `'0'` in the `memset()` was definitely bad though.

Reply:
We have a lot of such statements. I'm counting more than 50, most of them structs. Should I replace this anyway?

```
src/fd_device.c
41:	struct iovec iov = {0};
43:	struct msghdr msg = {0};

src/multicast_device.c
34:static mac_t ignore_src = {0};

src/process.c
48:static SERVICE_STATUS status = {0};

src/protocol.c
222:	past_request_t *new, p = {0};

src/protocol_edge.c
73:	sockaddr_t address, local_address = {0};

src/control.c
155:	sockaddr_t sa = {0};

src/splay_tree.c
28:	splay_node_t left = {0}, right = {0};

src/net.c
491:	signal_t sighup = {0};
492:	signal_t sigterm = {0};
493:	signal_t sigquit = {0};
494:	signal_t sigint = {0};
495:	signal_t sigalrm = {0};

src/ifconfig.c
91:	subnet_t address = {0};
174:	subnet_t subnet = {0}, gateway = {0};

src/node.c
149:	node_t n = {0};

src/linux/device.c
67:	struct ifreq ifr = {0};
119:		struct ifreq ifr_mac = {0};

src/openssl/prf.c
33:	digest_t digest = {0};

src/netutl.c
59:	struct addrinfo *ai, hint = {0};
79:	struct addrinfo *ai, hint = {0};
80:	sockaddr_t result = {0};

src/script.c
54:	static list_t list = {0};

src/route.c
122:	struct ip ip = {0};
123:	struct icmp icmp = {0};
224:	struct icmp6_hdr icmp6 = {0};

src/conf_net.c
5:	subnet_t subnet = {0};

src/sptps_test.c
327:	sockaddr_t sa = {0};

src/net_setup.c
640:	struct addrinfo *ai, hint = {0};

src/tincctl.c
680:	SERVICE_STATUS status = {0};
1851:			subnet_t s = {0};

src/protocol_subnet.c
48:	subnet_t s = {0}, *new, *old;
164:	subnet_t s = {0}, *find;

src/net_packet.c
1004:			node_id_t nullid = {0};
1710:	node_id_t nullid = {0};
1877:	sockaddr_t addr = {0};

test/unit/test_subnet.c
320:		subnet_t sub = {0};
352:	subnet_t sub = {0};
465:	const subnet_t sub = {0};
```

Reply:
Well, it passes the pre-existing CI checks, so I'm fine with the `{0}` then :)